liquid/liquid.go

```diff
@@ -0,0 +1,95 @@
+package liquid
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/VictoriaMetrics/fastcache"
+	"github.com/breez/server/breez"
+)
+
+func checkSwapId(fc *fastcache.Cache, apiURL, swapID string, body []byte) error {
+	c := &http.Client{Timeout: 2 * time.Second}
+	r, err := c.Get(apiURL + swapID)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+	if r.StatusCode != 200 {
+		return errors.New("bad swap id")
+	}
+	s := struct {
+		Status string `json:"status"`
+	}{}
+	err = json.NewDecoder(r.Body).Decode(&s)
+	if err != nil {
+		return err
+	}
+	if s.Status != "swap.created" {
```

Comment:
I think the status here could also be "invoice.set" for submarine swaps

Reply:
Yes, @yaslama Ross is right. invoice.set means the swapper has verified the payment info and ready for the lockup.